tests/Fixer/ControlStructure/NoUnneededNestedControlBlockFixerTest.php

```diff
@@ -0,0 +1,577 @@
+    /* 1 */ if (/* 2 */$bar)/* 3 */ { /* 4 */
+        echo 1;
+    }
+}
+',
+        ];
+
+        // precedence cases
+
+        yield '"if" case precedence "and"' => [
+            '<?php
+if (($foo and $c) && $bar) {
+
+
+
+}
+',
+            '<?php
+if ($foo and $c) {
+    if ($bar) {
+
+    }
+}
+',
+        ];
+
+        $precedenceCases = [
+            '$x && ($y %= $b)' => '$y %= $b',
+            '$x && ($y &= $b)' => '$y &= $b',
+            '$x && ($y **= $b)' => '$y **= $b',
+            '$x && ($y *= $b)' => '$y *= $b',
+            '$x && ($y += $b)' => '$y += $b',
+            '$x && ($y -= $b)' => '$y -= $b',
+            '$x && ($y .= $b)' => '$y .= $b',
+            '$x && ($y /= $b)' => '$y /= $b',
+            '$x && ($y <<= $b)' => '$y <<= $b',
+            '$x && ($y = $b)' => '$y = $b',
+            '$x && ($y >>= $b)' => '$y >>= $b',
+            '$x && ($y ? $a : $b)' => '$y ? $a : $b',
+            '$x && ($y ^= $b)' => '$y ^= $b',
+            '$x && ($y and $b)' => '$y and $b',
+            '$x && ($y or $b)' => '$y or $b',
+            '$x && ($y xor $b)' => '$y xor $b',
+            '$x && ($y |= $b)' => '$y |= $b',
+            '$x && ($y || $b)' => '$y || $b',
+        ];
+
+        foreach ($precedenceCases as $combined => $second) {
+            yield [
+                sprintf(
+                    '<?php
+function foo(){
+if (%s) {
+
+
+
+}
+}
+',
+                    $combined
+                ),
+                sprintf(
+                    '<?php
+function foo(){
+if ($x) {
+    if (%s) {
+
+    }
+}
+}
+',
+                    $second
+                ),
+            ];
+        }
+
+        yield 'already wrapped precedence' => [
+            '<?php
+if ($foo && $a && ($a xor $b)) {
+
+
+
+}
+',
+            '<?php
+if ($foo) {
+    if ($a && ($a xor $b)) {
+
+    }
+}
+',
+        ];
+
+        // multiple flat
+
+        $expected = str_repeat("if (\$foo && \$bar) {\n\n# 1\n\n}\n", 10);
+        $input = str_repeat("if (\$foo) {\n    if (\$bar) {\n# 1\n    }\n}\n", 10);
+
+        yield 'multiple flat "if" case' => [
+            "<?php\n".$expected,
+            "<?php\n".$input,
+        ];
+
+        // multiple nested
+
+        yield 'multiple "if" case' => [
+            '<?php
+if ($foo && $bar1 && $bar2 && $bar3) {
+
+
+
+
+
+
+
+}
+',
+            '<?php
+if ($foo) {
+    if ($bar1) {
+        if ($bar2) {
+            if ($bar3) {
+
+            }
+        }
+    }
+}
+',
+        ];
+
+        yield 'multiple "else" case' => [
+            '<?php
+if ($foo) {
+
+} else {
+    if ($bar) {
+
+    } else {
+        if ($bar1) {
+
+        } elseif ($bar2) {
+
+
+        }
+    }
+}
+',
+            '<?php
+if ($foo) {
+
+} else {
+    if ($bar) {
+
+    } else {
+        if ($bar1) {
+
+        } else {
+            if ($bar2) {
+            }
+        }
+    }
+}
+',
+        ];
+
+        yield 'multiple "elseif" case' => [
+            '<?php
+if ($foo) {
+
+} elseif($c && $bar) {
```

Comment:
The remaining `elseif` blocks are unneeded and should be removed as well.

Reply:
That would change the code behavior and make the fixer risky, also it out of scope of the rule.
For example;
```php
<?php

function foo() {
    echo 1;
}

if (false) {

} elseif (foo()) {

}
```
you reason all code can be removed here, yet `foo()` won't be called than and now it is

Reply:
Besides, having empty conditional block is a pretty rare use case and is typically covered by SCA already. Could be a nice new fixer though to remove such blocks.

Reply:
Right, my point is invalid regarding `elseif` but I think it's still valid for `else`.